bpo-41810: Reintroduce `types.EllipsisType`, `.NoneType` & `.NotImplementedType` (#22336)
python/cpython: Doc/library/constants.rst

.. data:: None

>    The sole value of the type :data:`types.NoneType`. ``None`` is frequently used to

Reviewer: "The sole value of the type NoneType" is most easily read (parsed) as referring to NoneType's usefulness.  I expected it to be followed by 'is to ...' and initially read 'is frequently' as a grammatical error.  It took awhile to realize that this is a convoluted  way to say 'None'.  I strongly recommend a rewording.

```suggestion
   ''None'' is the only instance of :data:`types.NoneType`.  It is frequently used to
```
`types.NoneType` is already twice identified as a type, so a third mention is not needed.

Author: Based on your and Guido's feedback (https://github.com/python/cpython/pull/22336#pullrequestreview-493014262) I'm proposing something like this:

```
.. data:: None

   An object frequently used to represent the absence of a value, as when
   default arguments are not passed to a function. Assignments to ``None``
   are illegal and raise a :exc:`SyntaxError`.
   Sole instance of the :data:`types.NoneType` type.
```
Any thoughts?

Reviewer 2: That's much better, I think. Although I'd make the last sentence (with the correct markup): "None is the sole instance of the NoneType type."

   represent the absence of a value, as when default arguments are not passed to a
   function. Assignments to ``None`` are illegal and raise a :exc:`SyntaxError`.

.. data:: NotImplemented

>    Sole value of the type :data:`types.NotImplementedType` and a
>    special value which should be returned by the binary special methods

Reviewer: See above.  Use a parallel construction.

```suggestion
   ``NotImplemented is the only instance of :data:`types.NotImplementedType`.  It is a
   special value which should be returned by the binary special methods
```

.. index:: single: ...; ellipsis literal
.. data:: Ellipsis

>    The same as the ellipsis literal "``...``" and sole value of the type

Reviewer: ```suggestion
   The same as the ellipsis literal "``...``" and the only instance of
```
